provider/common/availabilityzones.go

```diff
@@ -148,27 +151,45 @@ func AvailabilityZoneAllocations(env ZonedEnviron, ctx context.ProviderCallConte
 var internalAvailabilityZoneAllocations = AvailabilityZoneAllocations
 
 // DistributeInstances is a common function for implement the
-// state.InstanceDistributor policy based on availability zone
-// spread.
-func DistributeInstances(env ZonedEnviron, ctx context.ProviderCallContext, candidates, group []instance.Id) ([]instance.Id, error) {
-	// Determine the best availability zones for the group.
+// state.InstanceDistributor policy based on availability zone spread.
+func DistributeInstances(
+	env ZonedEnviron, ctx context.ProviderCallContext, candidates, group []instance.Id, limitZones []string,
+) ([]instance.Id, error) {
+	// Determine availability zone distribution for the group.
 	zoneInstances, err := internalAvailabilityZoneAllocations(env, ctx, group)
 	if err != nil || len(zoneInstances) == 0 {
 		return nil, err
 	}
 
+	// If there any zones supplied for limitation,
+	// filter to distribution data so that only those zones are considered.
+	filteredZoneInstances := zoneInstances[:0]
+	if len(limitZones) > 0 {
+		for _, zi := range zoneInstances {
+			for _, zone := range limitZones {
```

Comment:
here it would make a lot more sense to pass in a Set() or a map() so that we could do O(1) lookups. Though I suppose num zones isn't huge? (~10 is a lot?)

Reply:
Yeah, 10 zones seems a very loose constraint to me.

Fair enough notion though. I added it to the comment I made with your other suggestion, so that it can be done at such time as the input args become a struct.